src/cascade/input_data/db/locations.py

```diff
@@ -1,64 +1,115 @@
-from cascade.core.db import dbtrees
+import networkx as nx
 
+from cascade.core.db import db_queries
 from cascade.core.log import getLoggers
 
 CODELOG, MATHLOG = getLoggers(__name__)
 
 
-def get_location_hierarchy_from_gbd(execution_context):
-    return dbtrees.loctree(location_set_id=35, gbd_round_id=execution_context.parameters.gbd_round_id)
+def location_hierarchy(execution_context):
+    """
+    The GBD location hierarchy as a networkx graph where each node is the
+    location id, and its properties are all properties returned by
+    the dbtrees library. For instance
+
+    >>> locations = location_hierarchy(ec)
+    >>> assert locations.nodes[1]["level"] == 0
+    >>> assert locations.nodes[13]["location_name"] == "Malaysia"
+    >>> assert locations.successors(5) == [6, 7, 8]
+    >>> assert locations.predecessors(491) == [6]
+
+    Args:
+        execution_context: Uses the ``gbd_round_id``.
+
+    Returns:
+        nx.DiGraph: Each node is the location id as an integer.
+    """
+    location_df = db_queries.get_location_metadata(
+        location_set_id=35, gbd_round_id=execution_context.parameters.gbd_round_id)
+    G = nx.DiGraph()
+    G.add_nodes_from([(int(row.location_id), row._asdict()) for row in location_df.itertuples()])
+    # GBD encodes the global node as having itself as a parent.
+    G.add_edges_from([(int(row.parent_id), int(row.location_id))
+                      for row in location_df[location_df.location_id != 1].itertuples()])
+    G.graph["root"] = 1  # Global is the root location_id
+    return G
```

Comment:
This function is all of the hierarchies module. The only reason to add both nodes and edges, instead of just edges that imply nodes, is so that we can pick up all of the other columns from the incoming dataframe as node properties.